### DISCUSSION TITLE
Get P values for all Features from Seurat Object

### DISCUSSION THREAD

**Discussion by airfan0** [satijalab/seurat#5528] [Q&A]
Hi, I am trying to prepare a file for some pathway analysis using my results from Seurat. I was wondering if there was any way I could get the adjusted p-values Seurat calculates for all the features? I can find them for the differential expression genes after executing the FindAllMarkers function, but this only shows me the corrected p-values for selected features per identity class.

Thanks,
Amna

**Comment by samuel-marsh** [comment]
Hi Anna,

when running `FindAllMarkers` you need to adjust parameter thresholds the function normally uses for filtering the returned results.   You can also directly supply list of features and if you want all features you can just use `rownames(OBJECT_NAME)` for that.


Just as fyi if you are using p values to threshold for pathway analysis and you are running `FindAllMarkers` cluster vs cluster then the p values are inflated and as such not necessarily good to use as cutoff for such analysis.

Best,
Sam

**Reply by airfan0** [reply]
Hi Sam,

Thanks for the tip! Why do you say that the p-values are inflated and wouldn't be a good cut off for pathway analysis? In my project, I am trying to run a crosstalk analysis between various cell types. The cell types were identified via the standard workflow pipeline of Seurat. I was planning to take all the genes, their respective average expression values, and p-values for statistically significant per cell type as a data file input for the pathway analysis. What kind of p-value cut off would you recommend for crosstalk analysis between the various cell types?

Thanks,
Amna

**Reply by samuel-marsh** [reply]
So the reason is that pipelines for analysis in very basic terms: identify variable genes, use those genes to run PCA, then use that data to identify clusters in the same dataset.  So in short you identify differences in the data, use those differences to create cluster structure and then run DE test to identify what is different between clusters.

This section of the OSCA book addresses this.  There are some emerging publications and methods aiming to address these issues but from my view there isn't clear consensus yet as to the optimal method to do so.

In terms of identifying marker genes this type of DE is still fine.  It is often very clear just from plots of expression level and specificity of expressing cells whether or not a marker/set of markers is specific to given clusters.  But downstream analysis using thresholds gets tricky.

I think it your case it partially depends to how granular your clustering is/how different your cell types are.  If your cell types are very different then you can be fairly confident as to what is truly marker of that cell type.  However as cell types become more similar this becomes trickier and I don't know there is good answer as to what value to use.  Unfortunately for most part as well the larger the difference between cell types the less "useful" (in my opinion) pathway analysis truly is.  i.e. We know that for instance blood monocytes vs plasma cells are very different cells doing very different things and therefore how "useful" is the pathways analysis difference between those two populations?

Hope that makes sense.
Best,
Sam
